src/lib/validators.ts

```diff
@@ -21,6 +21,26 @@ export const updateGameSessionSchema = z.object({
   casualCapacity: z.number(),
 });
 
+export const insertSemesterSchema = createInsertSchema(semesters, {
+  name: z.string(),
+  startDate: z.coerce.date(),
+  endDate: z.coerce.date(),
+  breakStart: z.coerce.date(),
+  breakEnd: z.coerce.date(),
+  bookingOpenDay: z.enum(weekdayEnum.enumValues),
+  bookingOpenTime: z.string().time(),
+});
```

Comment:
You actually don't need to redefine alot of these fields as they are provided implicitly by the createInsertSchema function,

we only need to coerce the date as zod throws an error if it receives a date as a string, rather than a JS Date object. So we use this to first 'coerce' the string into a date type, and then parse if its valid

![image](https://github.com/UoaWDCC/uabc-portal/assets/119282717/1544ecca-5e28-44cd-bce1-8a43005e6246)